src/test/java/redmarket/commands/roll/RedMarketCommandTest.java

```diff
@@ -0,0 +1,90 @@
+package redmarket.commands.roll;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.javacord.api.DiscordApi;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.bbaker.discord.redmarket.commands.roll.RedMarketCommand;
+
+import de.btobastian.sdcf4j.CommandHandler;
+
+import redmarket.CommonMocks;
+
+public class RedMarketCommandTest extends CommonMocks {
+    
+    private CommandHandler commandHandler = null;
+    private DiscordApi api = null;
+    private RedMarketCommand cmd = null;
+
+    private Pattern validRollRgx;
+
+    @BeforeEach
+    public void setup() {
+        commandHandler = mock(CommandHandler.class);
+        api = mock(DiscordApi.class);
+        cmd = new RedMarketCommand(commandHandler);
+
+        validRollRgx = Pattern.compile(USER.getMentionTag() + ": black: \\d+ red: \\d+ \\`(Crit)?(Fail|Success)\\` Net: -?\\d+");
+    }
+
+    @Test
+    void testOnRoll() {
```

Comment:
You can keep this as is, but after it merges I will be converting it into a parametrized test. 

If you have the time and want to learn something new, checkout https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources

This is an excellent use case for ```
@ParameterizedTest
@ValueSource(strings = { "!r 1b ", "!r 2black", "!r 3r", "etc...." })```

Reply:
But I also want to complement the robust set of tests here. Very good.